rascal2/ui/view.py

```diff
@@ -21,11 +21,18 @@ def __init__(self):
         self.undo_view.setWindowIcon(window_icon)
         self.undo_view.setAttribute(QtCore.Qt.WidgetAttribute.WA_QuitOnClose, False)
 
+        self.presenter = MainWindowPresenter(self)
+
+        # TODO Remove this when the startup windows are added! just here to 'bootstrap' controls creation for the widget
```

Comment:
one thing to unpick here: the controls and project widget can't really initialise until the project exists - should their setup be split into two functions, one general `__init__` and a second setup which runs when setupMDI is run (which is after startup window closes (and so project exists) in #18)?

Reply:
i found the tidiest way was to move all controls-dependent code into a separate `setup_controls` function which runs when MDI is set up

Reply:
(in fact the startup dialog PR moves these widgets' creation into an `InitializeMDI` method so once that's merged this won't be necessary!)